graph/preprocessor_test.go

```diff
@@ -626,12 +626,112 @@ version:   'v1.1.0'`,
 version:   "v1.1.0'`,
 			"v1.1.0",
 		},
+		{
+			`       
+"version"`,
+			"",
+		},
+		{
+			` # task.yml file
+# should be skipped
+
+"version"  : v1.1.0    
+`,
+			"v1.1.0",
+		},
+		{
+			`
+build: something
+"version": v1.1.0`,
+			"",
+		},
+		{
+			`       
+"version"`,
+			"",
+		},
+		{
+			`       
+"version":v1.1.0
+			`,
+			"v1.1.0",
+		},
+		{
+			`       
+"version":foo:bar:beta`,
+			"foo:bar:beta",
+		},
+		{
+			`       
+"version":"v1.1.0"`,
+			"v1.1.0",
+		},
+		{
+			`       
+"version":   'v1.1.0'`,
+			"v1.1.0",
+		},
+		{
+			`       
+"version":   "v1.1.0'`,
```

Comment:
So from following this it looks like we end up supporting a fair bit of invalid inputs for parsing but since the goal of this function is to get the version, not validate that the yaml itself is valid I think this is fine